### PR TITLE
upcoming: [M3-8369] – Add "Encrypt Volume" checkbox in Edit Volume drawer

### PR DESCRIPTION
**Pull request by dwiley-akamai**
## Description 📝
Add a disabled "Encrypt Volume" checkbox in Edit Volume drawer that reflects the volume's encrypted status (i.e., encrypted volumes should have a checked disabled box, and unencrypted volumes should have an unchecked disabled box)

## Target release date 🗓️
9/3/24

## Preview 📷
<details>
<summary>Encrypted volume</summary>

Screenshot 2024-08-14 at 5 47 38 PM
</details>

<details>
<summary>Unencrypted volume</summary>

Screenshot 2024-08-14 at 5 47 29 PM

</details>

### Prerequisites
Point at the dev environment with the `blockstorage-encryption` tag on your account

### Verification steps
Confirm the Edit Volume drawer has the BSE checkbox when the feature flag is on, and that the checkbox reflects the volume's encrypted status (i.e., encrypted volumes should have a checked disabled box, and unencrypted volumes should have an unchecked disabled box)

## As an Author I have considered 🤔
- [X] 👀 Doing a self review
- [X] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [X] 🤏 Splitting feature into small PRs
- [X] ➕ Adding a changeset
- [X] 🧪 Providing/Improving test coverage
- [X] 🔐 Removing all sensitive information from the code and PR description
- [X] 🚩 Using a feature flag to protect the release
- [X] 👣 Providing comprehensive reproduction steps
- [X] 📑 Providing or updating our documentation
- [X] 🕛 Scheduling a pair reviewing session
- [X] 📱 Providing mobile support
- [X] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Volumes/EditVolumeDrawer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    server.use(
+      http.get(accountEndpoint, () => {
+        return HttpResponse.json(
+          accountFactory.build({ capabilities: ['Block Storage Encryption'] })
+        );
+      })
+    );
+
+    const { getByLabelText } = renderWithTheme(
+      <EditVolumeDrawer onClose={vi.fn} open={true} volume={volume} />,
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Volumes/EditVolumeDrawer.test.tsx:28]:
```suggestion
      <EditVolumeDrawer onClose={vi.fn} open volume={volume} />,
```

File: packages/manager/src/features/Volumes/EditVolumeDrawer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    server.use(
+      http.get(accountEndpoint, () => {
+        return HttpResponse.json(
+          accountFactory.build({ capabilities: ['Block Storage Encryption'] })
+        );
+      })
+    );
+
+    const { queryByRole } = renderWithTheme(
+      <EditVolumeDrawer onClose={vi.fn} open={true} volume={volume} />,
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Volumes/EditVolumeDrawer.test.tsx:52]:
```suggestion
      <EditVolumeDrawer onClose={vi.fn} open volume={volume} />,
```

File: packages/manager/src/features/Volumes/EditVolumeDrawer.test.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    const volume = volumeFactory.build();
+
+    server.use(
+      http.get(accountEndpoint, () => {
+        return HttpResponse.json(accountFactory.build({ capabilities: [] }));
+      })
+    );
+
+    const { queryByRole } = renderWithTheme(
+      <EditVolumeDrawer onClose={vi.fn} open={true} volume={volume} />,
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Volumes/EditVolumeDrawer.test.tsx:73]:
```suggestion
      <EditVolumeDrawer onClose={vi.fn} open volume={volume} />,
```

File: packages/manager/src/features/Volumes/EditVolumeDrawer.tsx [hunk 4]
```diff
@@ -114,6 +123,24 @@ export const EditVolumeDrawer = (props: Props) => {
...
         />
+        {isBlockStorageEncryptionFeatureEnabled && (
+          <Box
+            sx={{
+              marginLeft: '2px',
+              marginTop: '16px',
+            }}
+            alignItems="center"
+            display="flex"
+            flexDirection="row"
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Volumes/EditVolumeDrawer.tsx:134]: I don't think these 3 styled are nesseary

File: packages/manager/src/features/Volumes/EditVolumeDrawer.tsx [hunk 5]
```diff
@@ -114,6 +123,24 @@ export const EditVolumeDrawer = (props: Props) => {
...
+          >
+            <Checkbox
+              checked={volume?.encryption === 'enabled'}
+              disabled={true}
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Volumes/EditVolumeDrawer.tsx:138]:
```suggestion
              disabled
```

**Review by bnussman-akamai — APPROVED** [review]
Looks great!

**Review by mjac0bs — COMMENTED** [review]
Test coverage and the different states themselves look good!

I'm curious why we decided to go with a disabled checkbox here, rather than the lock icon we've been using elsewhere to denote encrypted, especially since this setting can't be changed.

Screenshot 2024-08-16 at 10 16 49 AM

**Comment by dwiley-akamai** [comment]
> I'm curious why we decided to go with a disabled checkbox here, rather than the lock icon we've been using elsewhere to denote encrypted, especially since this setting can't be changed.

I raised this with UX -- the disabled checkbox is more consistent with similar cases for LDE, and there's a chance that encrypting existing volumes will be supported down the line.

**Review by mjac0bs — APPROVED** [review]
> there's a chance that encrypting existing volumes will be supported down the line.

Got it. Thanks for the context!

**Pull request merged**